app/services/report_importer.rb

```diff
@@ -100,9 +100,12 @@ def notify_on_report_error(mail_error_state)
       end
 
       owners = host.owner.present? ? host.owner.recipients_for(:puppet_error_state) : []
+      users = MailNotification::PuppetError.all_hosts.flat_map(& :users)
```

Comment:
The scope `all_host` is needed, the `PuppetError` class has two distinct situations: all hosts subscription / owner host subscription. other solution is to generate another subclass.